src/dune_engine/targets.ml

```diff
@@ -263,6 +263,32 @@ module Produced = struct
     ;;
   end
 
+  (* Dummy digest because we want to continue discoverign all the errors
+     and we need some value to return in [mapi]. It will never be returned *)
+  let dummy_digest = Digest.generic ""
+
+  let collect_digests
+    { files; dirs }
+    ~(f : Path.Build.t -> 'a -> Cached_digest.Digest_result.t)
+    =
+    let errors = ref [] in
```

Comment:
We can get rid of `errors : _ list ref` and `dummy_digest` if we return `(_, _) either` from `f` and pipe it through a `List.partition_map ~f:Fn.id`

Reply:
Yup, but this means that we'll need create the intermediate list and then throw it away to create the map again. With the current code we:

1. Avoid an intermediate data structure
2. Re-create the map while relying on the existing sorted order

Given that this error case is quite error, it didn't seem right to make the whole code slower just for this.